test/test_spark_torch.py

```diff
@@ -0,0 +1,356 @@
+        metric12 = _generate_mock_metric('12', 12)
+        metric21 = _generate_mock_metric('21', 21)
+        metric22 = _generate_mock_metric('22', 22)
+
+        metric_value_groups = [[metric11, metric12], [metric21, metric22]]
+        all_metric_groups_values = get_metric_avgs(metric_value_groups)
+
+        assert all_metric_groups_values[0]['11'] == 11
+        assert all_metric_groups_values[0]['12'] == 12
+        assert all_metric_groups_values[1]['21'] == 21
+        assert all_metric_groups_values[1]['22'] == 22
+
+    def test_update_metrics(self):
+        def dummy_metric_add(output, label):
+            return output + label
+
+        def dummy_metric_sub(output, label):
+            return output - label
+
+        metric_fn_groups = [[dummy_metric_add, dummy_metric_sub], [dummy_metric_add]]
+
+        update_metrics = remote._update_metrics_fn(metric_fn_groups)
+
+        def _generate_mock_metric(name, val):
+            metric = mock.MagicMock()
+            metric.name = name
+            metric.avg.item.return_value = val
+            return metric
+
+        metric11 = _generate_mock_metric('11', 11)
+        metric12 = _generate_mock_metric('12', 12)
+        metric21 = _generate_mock_metric('21', 21)
+        metric22 = _generate_mock_metric('22', 22)
+
+        metric_value_groups = [[metric11, metric12], [metric21, metric22]]
+
+        outputs = [15, 4]
+        labels = [10, 2]
+
+        updated_metric_value_groups = update_metrics(metric_value_groups, outputs, labels)
+
+        updated_metric_value_groups[0][0].update.assert_called_once_with(25)
+        updated_metric_value_groups[0][1].update.assert_called_once_with(2)
+        updated_metric_value_groups[1][0].update.assert_called_once_with(25)
+        updated_metric_value_groups[1][1].update.assert_called_once_with(6)
+
+    def test_torch_param_serialize(self):
+        serialized_backend = _torch_param_serialize(EstimatorParams.backend.name, 'dummy_value')
+        assert serialized_backend is None
+
+        serialized_store = _torch_param_serialize(EstimatorParams.store.name, 'dummy_value')
+        assert serialized_store is None
+
+        serialized_dummy_param = _torch_param_serialize('dummy_param_name', None)
+        assert serialized_dummy_param is None
+
+    def test_torch_direct_parquet_train(self):
+        with spark_session('test_torch_direct_parquet_train') as spark:
+            df = create_xor_data(spark)
+
+            backend = CallbackBackend()
+            with local_store() as store:
+                store.get_train_data_path = lambda v=None: store._train_path
+                store.get_val_data_path = lambda v=None: store._val_path
+
+                with util.prepare_data(backend.num_processes(),
+                                       store,
+                                       df,
+                                       feature_columns=['features'],
+                                       label_columns=['y']):
```

Comment:
should we respect the verbose settings?

Reply:
You mean from pytest?  I am not sure of an easy way to do that, but happy to do so if you know how!

Reply:
Ah, I think you mean in the TorchEstimator.  Fixed.